benchmarks/src/main/java/zipkin2/it/ServerIntegratedBenchmark.java

```diff
@@ -0,0 +1,302 @@
+import org.testcontainers.containers.Network;
+import org.testcontainers.containers.wait.strategy.HttpWaitStrategy;
+import org.testcontainers.containers.wait.strategy.LogMessageWaitStrategy;
+import org.testcontainers.utility.MountableFile;
+import zipkin.server.ZipkinServer;
+
+class ServerIntegratedBenchmark {
+
+  static final Logger LOGGER = LoggerFactory.getLogger(ServerIntegratedBenchmark.class);
+
+  static final ObjectMapper OBJECT_MAPPER = new ObjectMapper();
+
+  @Nullable static final String RELEASED_ZIPKIN_VERSION = System.getenv("ZIPKIN_VERSION");
+
+  static final boolean WAIT_AFTER_BENCHMARK = "true".equals(System.getenv("ZIPKIN_BENCHMARK_WAIT"));
+
+  Closer closer;
+
+  @BeforeEach void setUp() {
+    closer = Closer.create();
+  }
+
+  @AfterEach void tearDown() throws Exception {
+    closer.close();
+  }
+
+  @Test void inMemory() throws Exception {
+    runBenchmark(null);
+  }
+
+  @Test void elasticsearch() throws Exception {
+    GenericContainer<?> elasticsearch = new GenericContainer<>("openzipkin/zipkin-elasticsearch7")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("elasticsearch")
+      .withLabel("name", "elasticsearch")
+      .withExposedPorts(9200)
+      .waitingFor(new HttpWaitStrategy().forPath("/_cluster/health"));
+    startContainer(elasticsearch);
+
+    runBenchmark(elasticsearch);
+  }
+
+  @Test void cassandra() throws Exception {
+    GenericContainer<?> cassandra = new GenericContainer<>("openzipkin/zipkin-cassandra")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("cassandra3")
+      .withLabel("name", "cassandra3")
+      .withExposedPorts(9042)
+      .waitingFor(new LogMessageWaitStrategy().withRegEx(".*Starting listening for CQL clients.*"));
+    startContainer(cassandra);
+
+    runBenchmark(cassandra);
+  }
+
+  @Test void mysql() throws Exception {
+    GenericContainer<?> mysql = new GenericContainer<>("openzipkin/zipkin-mysql")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("mysql")
+      .withLabel("name", "mysql")
+      .withExposedPorts(3306);
+    startContainer(mysql);
+
+    runBenchmark(mysql);
+  }
+
+  void runBenchmark(@Nullable GenericContainer<?> storage) throws Exception {
+    GenericContainer<?> zipkin = startZipkin(storage);
+
+    GenericContainer<?> backend = new GenericContainer<>("openzipkin/example-sleuth-webmvc")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("backend")
+      .withCommand("backend")
+      .withExposedPorts(9000)
+      .waitingFor(new HttpWaitStrategy().forPath("/actuator/health"));
+
+    GenericContainer<?> frontend = new GenericContainer<>("openzipkin/example-sleuth-webmvc")
+      .withNetwork(Network.SHARED)
+      .withNetworkAliases("frontend")
+      .withCommand("frontend")
+      .withExposedPorts(8081)
+      .waitingFor(new HttpWaitStrategy().forPath("/actuator/health"));
+
+    if (RELEASED_ZIPKIN_VERSION == null) {
+      Testcontainers.exposeHostPorts(9411);
```

Comment:
keep in mind that this feature starts an SSH port forwarding session and may affect the results of the benchmark (since traffic will be proxied, in not ultra performant way :D)

Reply:
Yeah it's definitely something to keep in mind, especially when comparing a docker version to in-process one. The numbers I'm seeing don't bother me too much.

Reply:
Realized I could copy the classpath into a Java ontainer, so now even for in-development code, the server is run in docker and the issue goes away. Having a method in testcontainers to start a container with the current classpath might be a very nice feature :)

Reply:
This sounds more like Arquillian :)

I am not sure such module will be easy to maintain because I expect a lot of hidden gems (module path alone makes me feel scared), but sounds like a nice idea for a library **on top** of Testcontainers 👍